tests/src/rules/no-restricted-paths.js

```diff
@@ -0,0 +1,87 @@
+import { RuleTester } from 'eslint'
+import rule from 'rules/no-restricted-paths'
+
+import { test, testFilePath } from '../utils'
+
+const ruleTester = new RuleTester()
+
+ruleTester.run('no-restricted-paths', rule, {
+  valid: [
+    test({
+      code: 'import foo from "../client/foo.js"',
+      filename: testFilePath('./restricted-paths/client/a.js'),
+      options: [
+        [ { target: './tests/files/restricted-paths/client', from: './tests/files/restricted-paths/server' } ],
+      ],
+    }),
+    test({
+      code: 'const foo = require("../client/foo.js")',
+      filename: testFilePath('./restricted-paths/client/a.js'),
+      options: [
+        [ { target: './tests/files/restricted-paths/client', from: './tests/files/restricted-paths/server' } ],
+      ],
+    }),
+  ],
```

Comment:
Mind adding some tests where the filename is in the `server` folder, for completeness' sake?